### PR TITLE
[virt-controller]: Small refactoring to avoid code duplication

### PR DESCRIPTION
**Pull request by iholder-redhat**
**What this PR does / why we need it**:
Very small refactoring to avoid duplicating the code (15 line +-) 3 times.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by acardace** [comment]
nice one!
/lgtm

File: pkg/virt-controller/watch/vm.go [hunk 1]
```diff
@@ -543,20 +543,26 @@ func (c *VMController) startStop(vm *virtv1.VirtualMachine, vmi *virtv1.VirtualM
 	}
 	log.Log.Object(vm).V(4).Infof("VirtualMachine RunStrategy: %s", runStrategy)
 
+	isStopRequestForVMI := func(vm *virtv1.VirtualMachine) bool {
+		if len(vm.Status.StateChangeRequests) != 0 {
```
Comment (ashleyschuett) [inline pkg/virt-controller/watch/vm.go:547]: I think below reads a bit better, but just a nit. 

```
if len(vm.Status.StateChangeRequests) == 0 {
    return false
}

stateChange := vm.Status.StateChangeRequests[0]

return stateChange.Action == virtv1.StopRequest &&
  stateChange.UID != nil &&
  *stateChange.UID == vmi.UID 
```

**Pull request merged**